### PR TITLE
INTMDB-512 l Private endpoint ADL

### PR DESCRIPTION
**Pull request by SuperMohit**
## Description

CFN resource for 
https://www.mongodb.com/docs/atlas/reference/api-resources-spec/#tag/Data-Federation-Private-Networks/operation/createOneDataFederationPrivateEndpointForOneProject

Link to any related issue(s): 

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Required Checklist:

- [ ] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added any necessary documentation (if appropriate)
- [ ] I have run `make fmt` and formatted my code

### CODE REVIEW

File: cfn-resources/private-endpoint-adl/README.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+Installation currently requires the follow 2 steps to build and then submit/register the
+new MongoDB::Atlas::PrivateEndpointADL Resource Type into your AWS Region. Note, this command uses the
+default AWS region.
+
+```bash
+TAGS=logging make
+cfn submit --verbose --set-default
+```
+## Cloudformation Examples
```
Comment (Zuhairahmed) [inline cfn-resources/private-endpoint-adl/README.md:42]:
```suggestion
## CloudFormation Examples
```

File: cfn-resources/private-endpoint-adl/README.md [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+Please see the [test/inputs_1_create.template.json](test/inputs_1_create.template.json) for example resource for example resource.
```
Comment (Zuhairahmed) [inline cfn-resources/private-endpoint-adl/README.md:44]: Can you move example to dedicated Examples folder: https://github.com/mongodb/mongodbatlas-cloudformation-resources/tree/master/examples

**Review by Zuhairahmed — APPROVED** [review]
Few minor comments, not blocking. Please resolve, otherwise LGTM

File: cfn-resources/private-endpoint-adl/cmd/resource/resource.go [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+	"github.com/aws-cloudformation/cloudformation-cli-go-plugin/cfn/handler"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/cloudformation"
+	"github.com/mongodb/mongodbatlas-cloudformation-resources/util"
+	"github.com/mongodb/mongodbatlas-cloudformation-resources/util/constants"
+	progressevents "github.com/mongodb/mongodbatlas-cloudformation-resources/util/progressevent"
+	"github.com/mongodb/mongodbatlas-cloudformation-resources/util/validator"
+	"go.mongodb.org/atlas/mongodbatlas"
+)
+
```
Comment (matt-condon) [inline cfn-resources/private-endpoint-adl/cmd/resource/resource.go:18]: Can we include logging on this resource for visibility?

**Review by matt-condon — APPROVED** [review]

**Review by rdharmedran — APPROVED** [review]

**Pull request merged**